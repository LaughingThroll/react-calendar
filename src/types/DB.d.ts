type TVacation = "Paid" | "UnPaid"

interface IVacation {
  startDate: string
  endDate: string
  type: TVacation
}

interface IMember {
  name: string
  vacations: IVacation[]
}

export interface ITeam {
  name: string
  percentageOfAbsent: number[]
  members: IMember[]
  teamId: number
}

export interface IDepartmentTeams {
  teams: ITeam[]
}
