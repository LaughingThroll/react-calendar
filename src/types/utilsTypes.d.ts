export type TChildren = string | JSX.Element | JSX.Element[]
