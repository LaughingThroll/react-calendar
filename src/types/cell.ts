export enum CellVariants {
  TH = 'th',
  TD = 'td',
}
