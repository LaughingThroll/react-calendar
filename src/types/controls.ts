export interface ControlForInput {
  value: string
  changeValue: (e: React.ChangeEvent<HTMLInputElement>) => void
}

export interface ControlForSelect {
  value: string
  changeValue: (e: React.ChangeEvent<HTMLSelectElement>) => void
}
