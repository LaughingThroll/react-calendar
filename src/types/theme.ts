export type TTheme = [string, string]
