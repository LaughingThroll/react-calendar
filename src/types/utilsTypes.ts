export type TChildren = string | JSX.Element | JSX.Element[]
export type ID = string
