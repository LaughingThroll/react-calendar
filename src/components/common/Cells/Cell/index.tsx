import React from 'react'
import classnames from 'classnames'

import styles from './../cells.module.scss'

import { isWeekend } from '../../../../utils/date'

import { CellVariants } from '../../../../types/cell'

export interface CellProps {
  cellDate?: Date
  variant?: CellVariants
  classNames?: string[]
  children?: any
}

const { cellGray } = styles

const Cell: React.FC<CellProps> = ({ variant = CellVariants.TD, children, cellDate, classNames = [] }) => {
  return React.createElement(
    variant,
    {
      className: `${classnames({
        [cellGray]: cellDate && isWeekend(cellDate),
      })}  ${classNames.join(' ')}`,
    },
    children
  )
}

export default React.memo(Cell)
