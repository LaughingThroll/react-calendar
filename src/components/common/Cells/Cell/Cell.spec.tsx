import { shallow } from 'enzyme'
import { CellVariants } from '../../../../types/cell'
import Cell, { CellProps } from './'

const setUp = (props: CellProps) => shallow(<Cell {...props} />)

describe('Cell render', () => {
  it('default render', () => {
    const props = { cellDate: new Date('2021-05-04') }
    const tree = setUp(props)

    expect(tree).toMatchSnapshot()
    expect(tree.find('td')).toHaveLength(1)
  })

  it('default render with class by default', () => {
    const props = { cellDate: new Date('2021-05-02') }
    const tree = setUp(props)

    expect(tree).toMatchSnapshot()
    expect(tree.find('td')).toHaveLength(1)
    expect(tree.find('.cellGray')).toHaveLength(1)
  })

  it('default render with variant', () => {
    const props = { cellDate: new Date('2021-05-02'), variant: CellVariants.TH }
    const tree = setUp(props)

    expect(tree).toMatchSnapshot()
    expect(tree.find('th')).toHaveLength(1)
  })
})
