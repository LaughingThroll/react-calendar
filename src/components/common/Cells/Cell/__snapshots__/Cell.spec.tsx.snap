// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell render default render 1`] = `
<td
  className="  "
/>
`;

exports[`Cell render default render with class by default 1`] = `
<td
  className="cellGray  "
/>
`;

exports[`Cell render default render with variant 1`] = `
<th
  className="cellGray  "
/>
`;
